src/Microsoft.Health.Fhir.SqlServer/Features/Search/CustomQueries.cs

```diff
@@ -0,0 +1,72 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Data;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Core;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search
+{
+    internal static class CustomQueries
+    {
+        private static DateTimeOffset _lastUpdatedQueryCache = DateTimeOffset.MinValue;
+        private static object lockObject = new object();
+        public static readonly ConcurrentDictionary<string, string> QueryStore = new ConcurrentDictionary<string, string>();
+
+        public static int WaitTime { get; set; } = 60;
+
+        public static string CheckQueryHash(IDbConnection connection, string hash, ILogger<SqlServerSearchService> logger)
+        {
+            var now = Clock.UtcNow;
```

Comment:
Thought, trim the seconds so all nodes query at roughly the same time? Or does this matter?

Reply:
I always thought that it is better to spread the calls to the database over whatever is acceptable. I hope we do not try to synchronize.

Reply:
I hope that this particular query is very low effort for SQL server.  So it does not make much of a difference either way.

Reply:
That is correct. I was just trying to make a point that we should be spreading calls, not synchronizing them.

Reply:
..in this case through, different nodes will have different behaviors until all have refreshed. Another use-case for a distributed cache.

Reply:
Distributed cache for this case sounds like overengineering for me.